rfcs/01391_async_sdk.md

```diff
@@ -0,0 +1,189 @@
+# RFC#01391: Oak Async SDK 
+
+# Objective
```

Comment:
Is this formatted according to the existing linters / checkers? Usually only one H1 header (the title) is allowed.

Reply:
Aah good point, I don't think it is (which is why tests are failing). I'll fix that, thanks!